tests/acceptance/features/apiSpaces/copySpaces.feature

```diff
@@ -451,3 +451,244 @@ Feature: copy file
+
+
+  Scenario: Copying a file to a folder with no permissions
+    Given using spaces DAV path
+    And user "Brian" has created folder "/testshare"
+    And user "Brian" has created a share with settings
+      | path        | testshare |
+      | shareType   | user      |
+      | permissions | read      |
+      | shareWith   | Alice     |
+    And user "Alice" has accepted share "/testshare" offered by user "Brian"
+    And user "Alice" has uploaded file with content "ownCloud test text file 0" to "/textfile0.txt"
+    When user "Alice" copies file "/textfile0.txt" from space "Personal" to "/testshare/textfile0.txt" inside space "Shares Jail" using the WebDAV API
+    Then the HTTP status code should be "403"
+    And user "Alice" should not be able to download file "/testshare/textfile0.txt" inside space "Shares Jail"
+
+
+  Scenario: Copying a file to overwrite a file into a folder with no permissions
+    Given using spaces DAV path
+    And user "Brian" has created folder "/testshare"
+    And user "Brian" has uploaded file with content "ownCloud test text file 1" to "textfile1.txt"
+    And user "Alice" has uploaded file with content "ownCloud test text file 0" to "/textfile0.txt"
+    And user "Brian" has created a share with settings
+      | path        | testshare |
+      | shareType   | user      |
+      | permissions | read      |
+      | shareWith   | Alice     |
+    And user "Alice" has accepted share "/testshare" offered by user "Brian"
+    And user "Brian" has copied file "textfile1.txt" to "/testshare/overwritethis.txt"
+    When user "Alice" copies file "/textfile0.txt" from space "Personal" to "/testshare/overwritethis.txt" inside space "Shares Jail" using the WebDAV API
+    Then the HTTP status code should be "403"
+    And for user "Alice" the content of the file "/testshare/overwritethis.txt" of the space "Shares Jail" should be "ownCloud test text file 1"
+
+
+  Scenario: copy a file over the top of an existing folder received as a user share
+    Given using spaces DAV path
+    And user "Alice" has uploaded file with content "ownCloud test text file 1" to "textfile1.txt"
+    And user "Brian" has created folder "/BRIAN-Folder"
+    And user "Brian" has created folder "BRIAN-Folder/sample-folder"
+    And user "Brian" has shared folder "BRIAN-Folder" with user "Alice"
+    And user "Alice" has accepted share "/BRIAN-Folder" offered by user "Brian"
+    When user "Alice" copies file "/textfile1.txt" from space "Personal" to "/BRIAN-Folder" inside space "Shares Jail" using the WebDAV API
+    Then the HTTP status code should be "204"
+    And for user "Alice" the content of the file "/BRIAN-Folder" of the space "Shares Jail" should be "ownCloud test text file 1"
+    And for user "Alice" folder "BRIAN-Folder" of the space "Shares Jail" should not contain these files:
+      | /sample-folder |
+    And as "Alice" file "/textfile1.txt" should exist
+    And user "Alice" should not have any received shares
+
```

Comment:
I understand this scenario but it's weird that you can overwrite the whole share with a file. should this be the expected behavior?

Reply:
This is exact same behavior as core.